probe/docker/registry_test.go

```diff
@@ -183,6 +187,13 @@ func TestRegistryEvents(t *testing.T) {
 		defer registry.Stop()
 		runtime.Gosched()
 
+		check := func(want []docker.Container) {
+			test.Poll(t, 10*time.Millisecond, func() bool {
+				have := allContainers(registry)
+				return reflect.DeepEqual(want, have)
+			}, "Missed event")
```

Comment:
We are missing `test.Diff(want, have)` , but I understand that it may not be possible in golang given how Poll is built.